specification/specification.tex

```diff
@@ -100,8 +100,48 @@ \subsubsection*{Spending Conditions}
 
 \subsection{Proxy Minting Validator}
 
-\subsection{Voting Effect Minting Validator}
+\subsection{Voting Effect Validator}
+
+The Voting Effect validator will utilise the above established Proxy Minting Validator, minting a Proxy Governance Authority Token (pGAT).
+By spending the pGAT, we essentially get the same guarantees as with GAT, but we can use Plutus V3, which introduces the functionalities to work with native Cardano governance.
+
+In order to vote on a Cardano Governance Action using this effect the following preconditions must be met:
+\begin{itemize}
+  \item The Voting Effect validator must be registered as a DRep
+  \item staking verification key of the DAO should be delegated to the above DRep
+  \item Governance action with well known id must be already submitted on-chain
+\end{itemize}
+
+The effect transaction will burn exactly one pGAT, and at the same time place a vote to the governance action defined in the datum.
+
+\subsection*{DRep registration}
+
+Registering the Voting Effect validator is a one time action, that has to be done after deployment of the protocol.
+The validator script will allow DRep registration unconditionally, but DRep update will be prohibited.
+
+\subsection*{Delegation to DRep}
+
+\subsection*{Data Types}
+
+\begin{lstlisting}[language=Haskell]
+data Datum = Datum
+  { governanceActionId :: GovernanceActionId
+  , vote :: Vote
+  }
+\end{lstlisting}
+
+\subsection*{Spending Conditions}
+
+\begin{itemize}
+  \item Transaction burns one PGAT (symbol is known from script parameter)
+  \item Spent UTxO contains PGAT
+  \item if \verb|ScriptPurpose| is \verb|Certifying| then the transaction passes unconditionally
+  \item if \verb|ScriptPurpose| is \verb|Voting|
+        \begin{itemize}
+          \item \verb|Voter| set to a \verb|DRepCertificate| with the script hash of this validator
+          \item \verb|Vote| in script context matches the one in the Datum
+        \end{itemize}
+\end{itemize}
```

Comment:
This needs to be still expanded a bit cause it gets problematic if you try to perform an attack by trying to use a single GAT to trigger multiple actions. And it's generally difficult to reason about cause making this script too liberal may compromise another component. Generally we want to forbid other stuff from happening in this tx like spending any other script-owned UTxO other than the one with the GAT, banning other governance actions, banning delegations, irrelevant certificates, mints, etc.